src/translator/translation_model.cpp

```diff
@@ -94,8 +95,10 @@ Ptr<Request> TranslationModel::makeRequest(size_t requestId, std::string &&sourc
   Segments segments;
   AnnotatedText annotatedSource;
 
+  HTML html(std::move(source), responseOptions.HTML);
```

Comment:
Response.source is modified by this constructor, and later both `source` and `target` are updated by `HTML::Restore`. However, these are both no-ops if `responseOptions.HTML` is `false`.

Reply:
It would be nicer and simpler to understand and maintain (in my opinion) if the guards happen here. i.e, HTML(...) is never constructed if  if `responseOptions.HTML` is `false`. Similarly for `html.Restore(...)`.  But feel free to prioritize important stuff ahead of this.

Reply:
Use std::optional?